File: 257-binary-tree-paths/binary-tree-paths.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def binaryTreePaths(self, root: Optional[TreeNode]) -> List[str]:
        if root == None: return []

        output = []

        def dfs(root):
            if not root:
                return
            out = []
            left = dfs(root.left)
            right = dfs(root.right)
            if left:
                for l in left:
                    out.append([str(root.val)] + l)
            if right:
                for r in right:
                    out.append([str(root.val)] + r)
            if len(out) == 0:
                return [[str(root.val)]]
            return out
        temp = dfs(root)
        for path in temp:
            if(len(path) == 1): output.append(str(path[0]))
            else:
                output.append("->".join(path))
        return output

File: 257-binary-tree-paths/binary-tree-paths.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {string[]}
 */
var binaryTreePaths = function(root) {
    let output = [], output2 = []

    const dfs = node => {
        if(!node) return []
        let left = dfs(node.left)
        let right = dfs(node.right)
        let temp = [...left, ...right]
        if(!temp.length){
            return [node.val]
        }
        for(let i = 0; i < temp.length; i++) {
            if(Array.isArray(temp[i])){
                temp[i] = [node.val, ...temp[i]]
            } else {
                temp[i] = [node.val, temp[i]]
            }
            
        }
        return temp
    }

    output = dfs(root)
    for(const o of output){
        if(Array.isArray(o)){
            output2.push(o.join("->"))
        } else {
            output2.push(String(o))
        }
        
    }
    return output2
};